Purchases/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -45,26 +59,58 @@ public typealias SK2Product = StoreKit.Product
         return hasher.finalize()
     }
 
+    @objc public var localizedDescription: String { self.product.localizedDescription }
+
+    @objc public var localizedTitle: String { self.product.localizedTitle }
+
+    @objc public var price: Decimal { self.product.price }
+
+    @objc public var localizedPriceString: String { self.product.localizedPriceString}
+
+    @objc public var productIdentifier: String { self.product.productIdentifier }
+
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 8.0, *)
+    @objc public var isFamilyShareable: Bool { self.product.isFamilyShareable }
+
+    @available(iOS 12.0, macCatalyst 13.0, tvOS 12.0, macOS 10.14, watchOS 6.2, *)
+    @objc public var subscriptionGroupIdentifier: String? { self.product.subscriptionGroupIdentifier}
+
+    @objc public var priceFormatter: NumberFormatter? { self.product.priceFormatter }
+
+    @available(iOS 11.2, macOS 10.13.2, tvOS 11.2, watchOS 6.2, *)
+    @objc public var subscriptionPeriod: SubscriptionPeriod? { self.product.subscriptionPeriod }
+
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, watchOS 6.2, *)
+    @objc public var introductoryPrice: PromotionalOffer? { self.product.introductoryPrice }
+
+    @available(iOS 12.2, macOS 10.14.4, tvOS 12.2, watchOS 6.2, *)
+    @objc public var discounts: [PromotionalOffer] { self.product.discounts }
+
+}
+
+/// Type that provides access to all of `StoreKit`'s product type's properties.
+internal protocol StoreProductType {
```

Comment:
This is explicitly `internal` because we don't need to expose it to users.
The docs do get inherited by `StoreProduct` so we can maintain that in only one place.